### PR TITLE
Simplify android compiling

### PR DESCRIPTION
**Pull request by m-kuhn**
Less boilerplate commands to build for android

- setup chainload toolchain automatically
- warn if ANDROID_NDK_HOME is not set
- enable WITH_VCPKG automatically

### CODE REVIEW

File: cmake/VcpkgToolchain.cmake [hunk 1]
```diff
@@ -9,15 +9,26 @@ string(ASCII ${NUGET_TOKEN_ASCII} NUGET_TOKEN_DEFAULT)
 set(NUGET_TOKEN ${NUGET_TOKEN_DEFAULT} CACHE STRING "Nuget token")
 
 string(COMPARE EQUAL "${CMAKE_HOST_SYSTEM_NAME}" "Windows" _HOST_IS_WINDOWS)
-set(WITH_VCPKG ${_HOST_IS_WINDOWS} CACHE BOOL "Use the vcpkg submodule for dependency management.")
+set(_WITH_VPKG_DEFAULT ${_HOST_IS_WINDOWS})
+if(${VCPKG_TARGET_TRIPLET} MATCHES "android")
```
Comment (m-kuhn) [inline cmake/VcpkgToolchain.cmake:13]:
```suggestion
if(${VCPKG_TARGET_TRIPLET} MATCHES "android")
  message(STATUS "-- Building for Android --")
```

Add status message

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e341e9804fa17f24af45134c55572c0fafe61029:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-2000cb0cc146ae28f1fa89a5b567e4a2e71d364a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-2000cb0cc146ae28f1fa89a5b567e4a2e71d364a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-2000cb0cc146ae28f1fa89a5b567e4a2e71d364a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-2000cb0cc146ae28f1fa89a5b567e4a2e71d364a-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d4a62c60e5f64e6485a2bd190620a47d5b05dafb:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-af9ae20eaf3532db1b684c176205224b17df5d91-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-af9ae20eaf3532db1b684c176205224b17df5d91-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-af9ae20eaf3532db1b684c176205224b17df5d91-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-af9ae20eaf3532db1b684c176205224b17df5d91-x86.apk)

**Review by nirvn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f91cda3d859bc75875100a7ab6188cf32ea65e6f:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-e92b3ab60f3856b5d4dbc7b736b463cdaa154f4e-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-e92b3ab60f3856b5d4dbc7b736b463cdaa154f4e-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-e92b3ab60f3856b5d4dbc7b736b463cdaa154f4e-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2336-e92b3ab60f3856b5d4dbc7b736b463cdaa154f4e-x86.apk)

**Pull request merged**